core/server/worker/src/main/java/alluxio/worker/block/management/ManagementTaskCoordinator.java

```diff
@@ -0,0 +1,182 @@
+    mLoadTracker = loadTracker;
+    mEvictionViewSupplier = evictionViewSupplier;
+
+    initializeTaskProviders();
+
+    // Read configs.
+    mLoadDetectionCoolDownMs =
+        ServerConfiguration.getMs(PropertyKey.WORKER_MANAGEMENT_LOAD_DETECTION_COOL_DOWN_TIME);
+    mIdleSleepMs = ServerConfiguration.getMs(PropertyKey.WORKER_MANAGEMENT_IDLE_SLEEP_TIME);
+
+    // Initialize management task executor.
+    // Currently a single management task is active at a time.
+    mTaskExecutor = Executors
+        .newSingleThreadExecutor(ThreadFactoryUtils.build("block-management-thread-%d", true));
+
+    // Initialize runner thread.
+    mRunnerThread = new Thread(this::runManagement, "block-management-runner");
+  }
+
+  /**
+   * Starts the coordinator.
+   */
+  public void start() {
+    // Start runner thread.
+    mRunnerThread.start();
+  }
+
+  /**
+   * Register known task providers by priority order.
+   *
+   * TODO(ggezer): TV2 - Implement pin enforcer as {@link BlockManagementTask}.
+   * TODO(ggezer): TV2 - Re-implement async-cache as {@link BlockManagementTask}.
+   */
+  private void initializeTaskProviders() {
+    mTaskProviders = new ArrayList<>(1);
+    if (ServerConfiguration.isSet(PropertyKey.WORKER_EVICTOR_CLASS)) {
+      LOG.warn("Tier management tasks will be disabled under eviction emulation mode.");
+    } else {
+      mTaskProviders.add(new TierManagementTaskProvider(mBlockStore, mMetadataManager,
+          mEvictionViewSupplier, mLoadTracker));
+    }
+  }
+
+  /**
+   * @return the next management task to run, {@code null} if none pending
+   */
+  private BlockManagementTask getNextTask() {
+    /**
+     * Order of providers in the registered list imposes an implicit priority of tasks.
+     * As long as a provider gives a task, providers next to it won't be consulted.
+     */
+    for (ManagementTaskProvider taskProvider : mTaskProviders) {
+      BlockManagementTask task = taskProvider.getTask();
+      if (task != null) {
+        return task;
+      }
+    }
+    // No task provided.
+    return null;
+  }
+
+  /**
+   * Main management loop.
+   */
+  private void runManagement() {
+    while (true) {
+      if (Thread.interrupted()) {
+        // Coordinator closed.
+        return;
+      }
+
+      BlockManagementTask currentTask = null;
+      try {
+        // Back off if any load detected.
+        // TODO(ggezer): TV2 - Trust management tasks' back-off handling.
+        if (mLoadTracker.loadDetected(BlockStoreLocation.anyTier())) {
+          LOG.info("Load detected.");
+          Thread.sleep(mLoadDetectionCoolDownMs);
+          continue;
+        }
+
+        final BlockManagementTask nextTask = getNextTask();
+        if (nextTask == null) {
+          LOG.info("No management task pending.");
+          Thread.sleep(mIdleSleepMs);
+          continue;
+        }
+
+        // Submit and wait for the task.
+        currentTask = nextTask;
+        mTaskExecutor.submit(() -> nextTask.run()).get();
```

Comment:
Add TODO here to run multiple tasks in parallel?

Reply:
At this level I don't intend to run things in parallel as each task here is I/O bound. I believe it's good to have priority based preemption here. Tasks themselves should spawn concurrent tasks though.

Reply:
If we don't plan to run tasks in parallel, do we still need to submit to another threadpool?

Reply:
It's a single-thread-executor. 
It can be useful to keep a separate thread pool for the future in case we want to let some tasks run in parallel if not all.

Reply:
Actually, after working on concurrency, I think it's simpler to run tasks on coordinator thread. We can change If we want to allow multiple tier management tasks later.